pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -214,35 +252,35 @@ func (m *mergeQuerier) mergeDistinctStringSlice(f stringSliceFunc) ([]string, st
 // Close releases the resources of the Querier.
 func (m *mergeQuerier) Close() error {
 	errs := tsdb_errors.NewMulti()
-	for pos, tenantID := range m.tenantIDs {
-		errs.Add(errors.Wrapf(m.queriers[pos].Close(), "failed to close querier for %s %s", rewriteLabelName(defaultTenantLabel), tenantID))
+	for pos, labelValue := range m.labelValues {
+		errs.Add(errors.Wrapf(m.queriers[pos].Close(), "failed to close querier for %s %s", rewriteLabelName(m.labelName), labelValue))
 	}
 	return errs.Err()
 }
 
 type selectJob struct {
-	pos      int
-	querier  storage.Querier
-	tenantID string
+	pos        int
+	querier    storage.Querier
+	labelValue string
 }
 
 // Select returns a set of series that matches the given label matchers. If the
-// tenantLabelName is matched on it only considers those queriers matching. The
+// `labelName` is matched on, it only considers those queriers matching. The
 // forwarded labelSelector is not containing those that operate on
-// tenantLabelName.
+// `labelName`.
 func (m *mergeQuerier) Select(sortSeries bool, hints *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
-	matchedTenants, filteredMatchers := filterValuesByMatchers(defaultTenantLabel, m.tenantIDs, matchers...)
+	matchedTenants, filteredMatchers := filterValuesByMatchers(m.labelName, m.labelValues, matchers...)
```

Comment:
is `matchedTenants` still correct here?
```suggestion
	matchedSeries, filteredMatchers := filterValuesByMatchers(m.labelName, m.labelValues, matchers...)
```